### PR TITLE
Sample rule

### PR DESCRIPTION
**Pull request by patrickkusebauch**
So, you cannot use the shim version if you want to use custom rules, see https://github.com/qossmic/deptrac/issues/916.

Therefore I converted it to a standard composer installation. PHAR should work fine too as you can see in the linked ticket. 

The rule itself is fairly simple, the only disadvantage is that you need to repeat the names of the PHP SPL exceptions inside the rule. Since they are well-known and fairly static, it does not seem too bad.

Basically
 - check that the reason for the dependency in the original case was an inheritance
 - check that the inheritance was from a PHP SPL library exception
 - If so, stop processing the event and don't count it as a dependency.

Feel free to ask me questions here.

### CODE REVIEW

**Comment by pckuijper** [comment]
Thanks for your input! This worked, I modified the script a little bit to make it work without magic strings and reference the classes directly and then also allow that specific class to have those dependencies as well.

Now I just have the challenge of implementing this inside our actual system, which is a bit tricky as we're still on an old version of symfony. But I guess using the PHAR will solve that issue then.
